### PR TITLE
feat: more human friendly makefile output prompts

### PR DESCRIPTION
**Pull request by leslie-tsang**
1. add makefile default target
2. more human friendly makefile output prompts
> before

```bash
Makefile rules:

    help:         Show Makefile rules.
    dev:          Create a development ENV
    test:         Run the test case
    clean:        Clean the test case
```
> after

```bash
[ info ] Makefile rules:

    make help                      : Show Makefile rules
    make dev                       : Create a development ENV
    make test                      : Run the test case
    make clean                     : Clean the test case
```
Signed-off-by: leslie tsang <leslie.tsang@icloud.com>

### CODE REVIEW

**Comment by spacewander** [comment]
Let's merge master to make CI pass.

**Comment by leslie-tsang** [comment]
> Let's merge master to make CI pass.

Done

**Review by membphis — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -1,41 +1,96 @@
...
-### help:         Show Makefile rules.
+# Makefile ENV
+ENV_OS_NAME ?= $(shell uname -s | tr '[:upper:]' '[:lower:]')
+ENV_RESTY   ?= LUA_PATH="$(LUA_PATH)" LUA_CPATH="$(LUA_CPATH)" resty
+
+# AWK patch for mawk
+ifneq ($(shell command -v gawk),)
+	ENV_HELP_AWK_RULE ?= '{ if(match($$0, /^\s*\#{3}\s*([^:]+)\s*:\s*(.*)$$/, res)){ printf("    make %-15s : %-10s\n", res[1], res[2]) } }'
+else
+	ENV_HELP_AWK_RULE := '{ if(match($$0, /^\#\#\#([^:]+):(.*)$$/)){ split($$0, res, ":"); gsub(/^\#\#\#[ ]*/, "", res[1]); _desc=$$0; gsub(/^\#\#\#([^:]+):[ \t]*/, "", _desc); printf("    make %-15s : %-10s\n", res[1], _desc) } }'
```
Comment (spacewander) [inline Makefile:21]: If I am you, I will prompt that people should use gawk instead. mawk is faster. But it doesn't support UTF8 well and one day you will find the support of mawk should be dropped.

**Review by spacewander — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (leslie-tsang) [reply Makefile:21]: Make sense, thanks for suggestion.

I add it for run test in container, I had change my mind. no necessary to increase maintenance costs

**Pull request merged**